app/graphql/types/base_object.rb

```diff
@@ -4,9 +4,25 @@ class BaseObject < GraphQL::Types::Relay::BaseObject
 
     connection_type_class Connections::BaseConnection
 
-    def self.timestamps
-      field :created_at, GraphQL::Types::ISO8601DateTime, null: true
-      field :updated_at, GraphQL::Types::ISO8601DateTime, null: true
+    class << self
+      def timestamps
+        field :created_at, GraphQL::Types::ISO8601DateTime, null: true
+        field :updated_at, GraphQL::Types::ISO8601DateTime, null: true
+      end
+
+      def has_many(name, type, null: true)
+        field name, type.connection_type, null: null,
+          resolve: proc { |object| CollectionLoader.for(object.class, name).load(object) }
+      end
+
+      def belongs_to(name, type, null: true)
+        field name, type, null: null,
+          resolve: proc { |object| RecordLoader.for(type.model_class).load(object.send("#{name}_id")) }
```

Comment:
Nothing we have to solve here, is it safe to assume the association id field is always generated by the `_id` suffix?

Reply:
I've found a better way. We can use `object.class.reflect_on_association(name).foreign_key`. WDYT?

Reply:
That's what I was trying to lead you to! ;-)